vips/vips.c

```diff
@@ -590,6 +590,10 @@ vips_pngsave_go(VipsImage *in, void **buf, size_t *len, int interlace, int quant
     else if (colors > 2) bitdepth = 2;
   } else {
     bitdepth = vips_get_palette_bit_depth(in);
+    if (bitdepth > 8) bitdepth = 16;
```

Comment:
TBH I'm not sure if `bitdepth > 8 && bitdepth != 16` is possible but just in case. Please let me know if I should remove this, @DarthSim . Cheers!

Reply:
The maximum palette size is 256, so the maximum `bitdepth` is 8. It's better to remove this to avoid errors.

Reply:
That makes total sense. It's updated now. Please have another look. Thanks!